### DISCUSSION TITLE
ApiCurio stackTrace in error response

### DISCUSSION THREAD

**Discussion by noako** [Apicurio/apicurio-registry#2140] [Q&A]
Hi, we are working with ApiCurio running on kubernetes
when calling the registry url with bad parameters (for example apis/registry/v2/groups/default/artifacts/123 where 123 doesn't exist)
the error response is huge and include stacktrace
I tried to set REGISTRY_API_ERRORS_INCLUDE_STACKTRACE and SCHEMA_REGISTRY_API_ERRORS_INCLUDE_STACKTRACE to false (even though it looks like it should be false by default), but for some reason this doesn't solve the issue.
is there a way to remove this data from the error response?
thanks

**Comment by EricWittmann** [comment]
Can you confirm what version of registry you are deploying?

**Reply by EricWittmann** [reply]
Note: the best way to do that is like this:

`curl http://localhost:8080/apis/registry/v2/system/info`  (sub the host and port for your registry obviously)

**Reply by noako** [reply]
{"name":"Apicurio Registry (Kafka+SQL)","description":"High performance, runtime registry for schemas and API designs.","version":"2.0.0.Final","builtOn":"2021-04-16T15:03:49+0000"}

**Reply by EricWittmann** [reply]
Ah OK that makes sense - the `REGISTRY_API_ERRORS_INCLUDE_STACKTRACE` option was introduced more recently.  The latest version is now 2.1.5.Final :: https://github.com/Apicurio/apicurio-registry/releases/tag/2.1.5.Final

**Reply by noako** [reply]
thanks, I will check if we can upgrade, is there a workaround that will acheive similar result in the version I have?
thanks

**Reply by EricWittmann** [reply]
I'm afraid there is nothing in the project itself that could help.  My only suggestion would be to put an API gateway/proxy in front of the registry that would allow you to modify the response.
